### PR TITLE
Teachingbubble a11y

### LINKED ISSUE #10503: [Accessibility][Keyboard Navigation]: Title of the TeachingBubble dialog is not defined as heading. ("Bug2")
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B2f27d6ee-4e41-49bd-8216-6ca26df23f17%7D&action=default&uid=%7B2F27D6EE-4E41-49BD-8216-6CA26DF23F17%7D&ListItemId=1837&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble)

This issue represents an issue from #8088

### PR DESCRIPTION
**Pull request by jdhuntington**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #10503
- [X] Include a change request file using `$ yarn change`

#### Description of changes

Specify aria role for teaching bubble's heading.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10511)

### CODE REVIEW

**Comment by jdhuntington** [comment]
Using an `h*` element would feel more correct here, but there are 2 problems:
* we can't infer the correct heading level
* existing style rules might target `div` or `p` tags

In light of these, left the tag names the same and added a role only.

File: packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubbleContent.base.tsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ export class TeachingBubbleContentBase extends BaseComponent<ITeachingBubbleProp
 
       headerContent = (
         <div className={classNames.header}>
-          <HeaderWrapperAs className={classNames.headline} id={ariaLabelledBy}>
+          <HeaderWrapperAs role="heading" className={classNames.headline} id={ariaLabelledBy}>
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubbleContent.base.tsx]: you can specify level as well

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role

Comment (jdhuntington) [reply packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubbleContent.base.tsx]: Works for me. I'm not sure why it's better though...

**Comment by micahgodbolt** [comment]
oh shoot...i saw 'bug2' on the original issue and that stuck in my head as 'heading2'. I thought the original request was for a level 2 heading. Yeah, disregard.

**Comment by AlexHenkel** [comment]
little late to discussion but our team is facing an a11y issue due to a missing `aria-level` attribute, since we run an [Axe scan](https://www.deque.com/axe/) on our CI pipeline. I agree with @jdhuntington on "no aria is better than bad aria", but in our case this blocks us completely from using the `TeachingBubble` component as this will get always flagged for us. So I would appreciate your thoughts on adding an optional prop to set `aria-label` here that has undefined as fallback, so it's not set automatically set. CC @micahgodbolt 

Attaching screenshot of Axe scan on Fluent UI demo page
image

**Pull request merged**